tests/test_util/test_warning.py

```diff
@@ -1,38 +1,88 @@
 #!/usr/bin/env python
 
-"""Unit tests for union_dict.
-"""
-from unittest import TestCase, main
+import pickle
 
-from cogent3.util.warning import deprecate
+import pytest
 
+from cogent3.util.warning import deprecated_args
 
-__author__ = "Richard Morris"
-__copyright__ = "Copyright 2007-2022, The Cogent Project"
-__credits__ = ["Gavin Huttley", "Richard Morris"]
-__license__ = "BSD-3"
-__version__ = "2023.2.12a1"
-__maintainer__ = "Gavin Huttley"
-__email__ = "gavin.huttley@anu.edu.au"
-__status__ = "Production"
 
+def test_deprecated_args():
```

Comment:
you'll want to trap the deprecation warning in this test, otherwise there'll be too much guff displayed after the test run.

Since testing the correctness of returned value is the purpose here, I suggest using the [@pytest.mark.filterwarnings](https://docs.pytest.org/en/7.1.x/how-to/capture-warnings.html#pytest-mark-filterwarnings) decorator.